### PR TITLE
feat: make partition calculation parallel

### PR DESCRIPTION
**Pull request by chebbyChefNEQ**
make IVF partition calculation parallelize

### CODE REVIEW

File: rust/lance-index/src/vector/ivf.rs [hunk 1]
```diff
@@ -290,18 +292,42 @@ impl<T: ArrowFloatType + Dot + L2 + Cosine + 'static> IvfImpl<T> {
...
-        .in_current_span()
-        .await
-        .expect("compute_partitions: schedule CPU task")
+        let num_centroids = centroids.len() / dimension;
+        let num_rows = data.len() / dimension;
+
+        let chunks = std::cmp::min(num_cpus::get(), num_rows);
+
+        info!(
+            "computing parition on {} chunks, out of {} centroids, and {} vectors",
```
Comment (wjones127) [inline rust/lance-index/src/vector/ivf.rs:301]:
```suggestion
            "computing partition on {} chunks, out of {} centroids, and {} vectors",
```

File: rust/lance/src/index/vector/ivf.rs [hunk 2]
```diff
@@ -714,6 +714,8 @@ pub async fn build_ivf_pq_index(
             vec![],
             None,
         )?;
+
+        info!("starting to compute partitions for PQ training, sample size");
```
Comment (wjones127) [inline rust/lance/src/index/vector/ivf.rs:718]: Did you mean to log the sample size or something?

Comment (chebbyChefNEQ) [reply rust/lance/src/index/vector/ivf.rs:718]: ah yes, fixing

**Review by rok — APPROVED** [review]

File: rust/lance-index/src/vector/ivf.rs [hunk 3]
```diff
@@ -290,18 +292,42 @@ impl<T: ArrowFloatType + Dot + L2 + Cosine + 'static> IvfImpl<T> {
...
+            chunks, num_centroids, num_rows,
```
[the remark this answers is not in the document]
Comment (chebbyChefNEQ) [reply rust/lance-index/src/vector/ivf.rs:302]: added `as usize` to `chunks`. the other one are already usize and clippy complains

**Review by wjones127 — APPROVED** [review]

**Pull request merged**